File: mac/Classes/Model/Results/LRMessage.m

#import "LRMessage.h"


@interface LRMessage ()

@end


@implementation LRMessage

- (instancetype)initWithSeverity:(LRMessageSeverity)severity text:(NSString *)text filePath:(NSString *)filePath line:(NSInteger)line column:(NSInteger)column {
    self = [super init];
    if (self) {
        _severity = severity;
        _text = [text copy];
        _filePath = [filePath copy];
        _line = line;
        _column = column;
    }
    return self;
}

- (NSString *)description {
    return [NSString stringWithFormat:@"%@ in %@:%d:%d: %@", (_severity == LRMessageSeverityError ? @"Error" : @"Warning"), _filePath, (int)_line, (int)_column, _text];
}

@end

File: mac/Classes/Model/Results/LRMessage.swift

import Foundation

@objc class LRMessage : NSObject, Printable {
    let severity : LRMessageSeverity;
    let text : String;
    let filePath : String?;
    let line: Int;
    let column: Int;

    var stack : String = "";
    var rawOutput : String = "";

    init(severity: LRMessageSeverity, text: String, filePath: String?, line: Int, column: Int) {
        self.severity = severity;
        self.text = text;
        self.filePath = filePath;
        self.line = line;
        self.column = column;
    }

    override var description : String {
        let severityMessage = (severity == LRMessageSeverity.Error ? "Error" : "Warning")
        return "\(severityMessage) in \(filePath):\(line):\(column): \(text)"
    }
}
